contracts/contracts/token/OUSD.sol

```diff
@@ -208,229 +220,164 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
         address _from,
         address _to,
         uint256 _value
-    ) public override returns (bool) {
+    ) external returns (bool) {
         require(_to != address(0), "Transfer to zero address");
-        require(_value <= balanceOf(_from), "Transfer greater than balance");
+        require(_value <= allowances[_from][msg.sender], "Allowance exceeded");
 
-        _allowances[_from][msg.sender] = _allowances[_from][msg.sender].sub(
-            _value
-        );
+        allowances[_from][msg.sender] -= _value;
```

Comment:
A good practice is to keep the allowance at maximum value if it's set to max value, it saves some gas avoiding a write.